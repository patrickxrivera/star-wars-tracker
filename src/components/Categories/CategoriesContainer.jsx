import React, { Component } from 'react';
import styled from 'styled-components';
import icons from './CategoryData.jsx';
import { Container, CategoryStyles, CategoryName } from './CategoryStyles.jsx';

const Category = styled.div`
  ${CategoryStyles};
`;

class CategoriesContainer extends Component {
  constructor(props) {
    super(props);

    [
      'handleClick',
      'getJSXForCategories',
      'getIconDataFrom',
      'createJSXFrom'
    ].forEach((propToBind) => {
      this[propToBind] = this[propToBind].bind(this);
    });
  }

  handleClick() {
    return null;
  }

  getJSXForCategories() {
    const categoriesJSX = Object.keys(icons).map((categoryName) => {
      const iconData = this.getIconDataFrom(icons, categoryName);
      const categoryJSX = this.createJSXFrom(iconData);
      return categoryJSX;
    });

    return categoriesJSX;
  }

  getIconDataFrom(icons, categoryName) {
    const [color, Icon] = [
      icons[categoryName].color,
      icons[categoryName].iconName
    ];
    return { color, Icon, categoryName };
  }

  createJSXFrom({ color, Icon, categoryName }) {
    return (
      <Category
        onClick={this.handleClick}
        key={categoryName}
        color={color}
        data-test={`category-${categoryName}`}
      >
        <Icon size={35} />
        <CategoryName>{categoryName}</CategoryName>
      </Category>
    );
  }

  render() {
    const categoriesJSX = this.getJSXForCategories();

    return <Container>{categoriesJSX}</Container>;
  }
}

export default CategoriesContainer;
