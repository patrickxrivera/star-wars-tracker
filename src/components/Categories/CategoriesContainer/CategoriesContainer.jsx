import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import iconData from './CategoriesContainerData.jsx';
import Category from '../Category/Category.jsx';
import Container from './CategoriesContainerStyles.jsx';
import ResultsContainer from '../../Results/ResultsContainer/ResultsContainer.jsx';

class CategoriesContainer extends Component {
  constructor(props) {
    super(props);

    this.state = {
      selected: null
    };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(selectedCategory) {
    this.setState(() => {
      return { selected: selectedCategory };
    });
  }

  render() {
    const selected = this.state.selected;

    const categories = Object.values(iconData).map(
      ({ categoryName, color, Icon }) => {
        return (
          <Category
            categoryName={categoryName}
            color={color}
            Icon={Icon}
            handleClick={this.handleClick}
            key={categoryName}
          />
        );
      }
    );

    return (
      <div>
        <Container>{categories}</Container>
        {!!selected && <Link to={{ pathname: `results` }} />};
      </div>
    );
  }
}

export default CategoriesContainer;
