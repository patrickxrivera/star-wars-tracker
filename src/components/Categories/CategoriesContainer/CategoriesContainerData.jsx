import { User, Chrome, Send } from 'react-feather';

const iconData = {
  1: {
    categoryName: 'People',
    color: '#4f5ef5',
    Icon: User
  },
  2: {
    categoryName: 'Planets',
    color: '#f44c67',
    Icon: Chrome
  },
  3: {
    categoryName: 'Vehicles',
    color: '#2fca74',
    Icon: Send
  }
};

export default iconData;
