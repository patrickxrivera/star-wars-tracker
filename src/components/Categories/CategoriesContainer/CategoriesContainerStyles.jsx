import styled from 'styled-components';
import { Link } from 'react-router-dom';

const Container = styled.div`
  display: flex;
  flex: 1;
  justify-content: center;
  margin-top: ${(props) => props.theme.main};
`;

const StyledLink = styled(Link)`
  text-decoration: none;

  &:focus,
  &:hover,
  &:visited,
  &:link,
  &:active {
    text-decoration: none;
  }
`;

Container.defaultProps = {
  theme: {
    main: '4.25rem'
  }
};

export { Container, StyledLink };
