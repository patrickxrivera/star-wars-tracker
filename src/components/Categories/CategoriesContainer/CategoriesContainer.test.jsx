import React from 'react';
import { mount } from 'enzyme';

import CategoriesContainer, { Category } from './CategoriesContainer.jsx';

test('CategoriesContainer renders correctly', () => {
  const wrapper = mount(<CategoriesContainer />);
  expect(wrapper).toMatchSnapshot();
});
