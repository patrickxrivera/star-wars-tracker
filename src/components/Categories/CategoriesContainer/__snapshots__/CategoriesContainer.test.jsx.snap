// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesContainer renders correctly 1`] = `
<CategoriesContainer>
  <styled.div>
    <div
      className="sc-bdVaJa jvHuTm"
    >
      <Category
        categoryName="People"
        color="#4f5ef5"
        data-test="category-People"
        icon={[Function]}
        key="People"
        onClick={[Function]}
      >
        <styled.div
          color="#4f5ef5"
          data-test="category-People"
          key="People"
        >
          <div
            className="sc-htpNat kCQikd"
            color="#4f5ef5"
            data-test="category-People"
          >
            <User
              color="currentColor"
              size={35}
            >
              <svg
                fill="none"
                height={35}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={35}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="12"
                  cy="7"
                  r="4"
                />
              </svg>
            </User>
            <styled.span>
              <span
                className="sc-bwzfXH ggUUZa"
              >
                People
              </span>
            </styled.span>
          </div>
        </styled.div>
      </Category>
      <Category
        categoryName="Planets"
        color="#f44c67"
        data-test="category-Planets"
        icon={[Function]}
        key="Planets"
        onClick={[Function]}
      >
        <styled.div
          color="#f44c67"
          data-test="category-Planets"
          key="Planets"
        >
          <div
            className="sc-htpNat fohWqW"
            color="#f44c67"
            data-test="category-Planets"
          >
            <Chrome
              color="currentColor"
              size={35}
            >
              <svg
                fill="none"
                height={35}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={35}
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="10"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="4"
                />
                <line
                  x1="21.17"
                  x2="12"
                  y1="8"
                  y2="8"
                />
                <line
                  x1="3.95"
                  x2="8.54"
                  y1="6.06"
                  y2="14"
                />
                <line
                  x1="10.88"
                  x2="15.46"
                  y1="21.94"
                  y2="14"
                />
              </svg>
            </Chrome>
            <styled.span>
              <span
                className="sc-bwzfXH ggUUZa"
              >
                Planets
              </span>
            </styled.span>
          </div>
        </styled.div>
      </Category>
      <Category
        categoryName="Vehicles"
        color="#2fca74"
        data-test="category-Vehicles"
        icon={[Function]}
        key="Vehicles"
        onClick={[Function]}
      >
        <styled.div
          color="#2fca74"
          data-test="category-Vehicles"
          key="Vehicles"
        >
          <div
            className="sc-htpNat kUdVuO"
            color="#2fca74"
            data-test="category-Vehicles"
          >
            <Send
              color="currentColor"
              size={35}
            >
              <svg
                fill="none"
                height={35}
                stroke="currentColor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 24 24"
                width={35}
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="22"
                  x2="11"
                  y1="2"
                  y2="13"
                />
                <polygon
                  points="22 2 15 22 11 13 2 9 22 2"
                />
              </svg>
            </Send>
            <styled.span>
              <span
                className="sc-bwzfXH ggUUZa"
              >
                Vehicles
              </span>
            </styled.span>
          </div>
        </styled.div>
      </Category>
    </div>
  </styled.div>
</CategoriesContainer>
`;
